### DISCUSSION TITLE
CacheCoverage 0 days | Memory is overutilized, so cooling down segment

### DISCUSSION THREAD

**Discussion by hbre** [datalust/seq-tickets#1171] [Help]
Hi,

I have two different windows 2019 servers with plenty of RAM (256GB). I installed the current versions (updated today).

But... Both machines don't seem to cache anything:  **Cache coverage 0 ms**

I found something about `Memory is overutilized, so cooling down segment` in the internal logs?

Any ideas what may be my configuration mistake? 

# machine 1
seq is configured with a RAM-Target of 0.25. I would expect seq to try to use 64GB of Memory. 
```powershell
PS C:\Windows\system32> seq config -k cache.systemRamTarget
0.25
```

![image](https://user-images.githubusercontent.com/898832/111149715-8436d100-858d-11eb-93de-6c90669c5822.png)

via `/settings/diagnostics` I see

* RAM cache coverage | 0 days
* RAM cache size | 0 events
* Process working set | 643 MB

# machine 2
Another machine of mine (develop) is configured with 32GB systemRamTarget, but the metrics are nearly the same.

```powershell
PS C:\Windows\system32> seq config -k cache.systemRamTarget
34359738368
```

Both machines don't have the `SEQ_CACHE_SYSTEMRAMTARGET` Variable.

# seq Internal log

```json
{"@t":"2021-03-15T12:06:18.5316840Z","@mt":"Seq {SeqVersion} running on OS {OSCaption}","SeqVersion":"2021.1.5425","OSCaption":"Microsoft Windows Server 2019 Standard","SourceContext":"Seq.Server.Features.Diagnostics.EnvironmentInformationWriter"}
{"@t":"2021-03-15T12:06:18.6106923Z","@mt":"Seq detected {TotalRAM} MB of RAM","TotalRAM":274512.785408,"SourceContext":"Seq.Server.Features.Diagnostics.EnvironmentInformationWriter"}
{"@t":"2021-03-15T12:06:18.6388354Z","@mt":"Running in Windows Service host"}
{"@t":"2021-03-15T12:06:20.4811248Z","@mt":"Opening event store at {StreamPath}","StreamPath":"F:\\Seq\\Stream\\stream.flare","SourceContext":"Flare.Cluster.StorageNodeCoordinator"}
{"@t":"2021-03-15T12:06:20.5810246Z","@mt":"Opening metastore {MetastorePath}","MetastorePath":"F:\\Seq\\Documents\\default.quince","InstanceName":"Quince","SourceContext":"Quince.Storage.Esent.EsentDocumentStore"}
{"@t":"2021-03-15T12:06:20.6076937Z","@mt":"Cache warm-up is required","SourceContext":"Flare.Cache.SegmentCache","OperationId":"76ef7cce-f8d5-4aa3-b0a9-595135a09b7d"}
{"@t":"2021-03-15T12:06:20.6167661Z","@mt":"Initial memory cache warm-up {Outcome} in {Elapsed:0.0} ms","@r":["9.1"],"Outcome":"completed","Elapsed":9.0813,"OperationId":"76ef7cce-f8d5-4aa3-b0a9-595135a09b7d"}
{"@t":"2021-03-15T12:06:20.6232619Z","@mt":"{WarmSegments} segments warmed up ({@Shared})","WarmSegments":0,"Shared":{"Schemata":0,"Strings":0},"SourceContext":"Flare.Cache.SegmentCache"}
{"@t":"2021-03-15T12:06:20.7171418Z","@mt":"Storage subsystem available"}
{"@t":"2021-03-15T12:06:20.8789724Z","@mt":"Seq listening on {ListenUris}","ListenUris":["https://XXX:5341/"]}
{"@t":"2021-03-15T12:06:51.0899620Z","@mt":"Memory is overutilized, so cooling down segment {Segment} ({HotCount} events)","Segment":"2021-03-15T12:00:00.0000000Z","HotCount":16332,"SourceContext":"Flare.Cache.SegmentCache"}
```

**Comment by nblumhardt** [comment]
Hi Henning,

Thanks for getting in touch. The `cache.systemRamTarget` doesn't work the way you might expect - it specifies a _total system memory utilization threshold_ that isn't specific to the Seq process.

It's designed to ensure there's enough headroom on the server to handle load/traffic spikes without exhausting all available memory.

If the value is 0.25, then when the sum of memory used by all processes is above 0.25, Seq will get out the way.

Seq doesn't share well with others - in production, it does best on its own. But running Seq with the default memory target should work well in this case as long as SQL Server's RAM consumption is limited:

```
seq config clear -k cache.systemRamTarget
seq service restart
```

Let me know if this helps,
Nick
